packages/vats/test/name-hub-published.test.js

```diff
@@ -58,3 +61,30 @@ test('promise space reserves non-well-known names', async t => {
 
   t.is(await nameHub.lookup('thing1'), true);
 });
+
+test('promise space reset instance to change a value', async t => {
+  const { nameAdmin, nameHub: agoricNames } = makeNameHubKit();
+
+  const tracer = makeTracer('test NS');
+  const space = await makeWellKnownSpaces(
+    nameAdmin,
+    tracer,
+    Object.keys(agoricNamesReserved),
+  );
+
+  const auctionInstance1 = makeHandle('instance1');
+  const instanceLookup = await agoricNames.lookup('instance');
+
+  // @ts-expect-error storing fake instance
+  space.instance.produce.auctioneer.resolve(auctionInstance1);
+  // @ts-expect-error expecting instance
+  t.is(await space.instance.consume.auctioneer, auctionInstance1);
```

Comment:
I suggest moving the error to where `auctionInstance1` is declared:

```suggestion
  /** @type { Instance } */
  // @ts-expect-error mock
  const auctionInstance1 = makeHandle('instance1');
  const instanceLookup = await agoricNames.lookup('instance');

  space.instance.produce.auctioneer.resolve(auctionInstance1);
  t.is(await space.instance.consume.auctioneer, auctionInstance1);
```